src/traits.rs

```diff
@@ -215,26 +330,245 @@ pub trait Context {
 }
 
 pub trait RootContext: Context {
+    /// Called when the host starts the WebAssembly Virtual Machine.
+    ///
+    /// Its configuration (of `_vm_configuration_size`) can be retrieved using `self.get_vm_configuration()`.
+    ///
+    /// # Arguments
+    ///
+    /// * `vm_configuration_size` - the size of the VM configuration
+    ///
+    /// # Returns
+    ///
+    /// * `bool` - `true` if the configuration was processed successfully, `false` otherwise
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// use proxy_wasm::traits::*;
+    /// use proxy_wasm::types::*;
+    /// use log::debug;
+    ///
+    /// struct MyRootContext;
+    ///
+    /// #[derive(serde::Deserialize, Debug)]
+    /// struct MyVmConfiguration {
+    ///     /// Some key
+    ///     pub key: String,
+    /// }
+    ///
+    /// impl RootContext for MyRootContext {
+    ///   fn on_vm_start(&mut self, _vm_configuration_size: usize) -> bool {
+    ///     let vm_configuration = self.get_vm_configuration().unwrap();
+    ///
+    ///     let parsed_vm_configuration: MyVmConfiguration =
+    ///         serde_json::from_slice::<MyVmConfiguration>(&vm_configuration).unwrap();
+    ///
+    ///     // Do something with the parsed vm configuration
+    ///     debug!("vm_configuration: {:?}", parsed_vm_configuration);
+    ///
+    ///     true
+    ///   }
+    /// }
+    ///
+    /// # impl Context for MyRootContext {}
```

Comment:
I was wondering if the `Context`-implementations, which are needed for the successful compilation, should be shown or hidden in the examples. @PiotrSikora